File: apk_monytix/app/src/main/java/com/example/apk/console/tabs/AccountsTabView.kt
package com.example.apk.console.tabs

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.runtime.getValue
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.example.apk.console.MolyConsoleViewModel
import com.example.apk.ui.components.GlassCard
import com.example.apk.ui.theme.Gold
import com.example.apk.ui.theme.TextPrimary
import com.example.apk.ui.theme.TextSecondary
import com.example.apk.console.utils.formatCurrency

@Composable
fun AccountsTabView(viewModel: MolyConsoleViewModel) {
    val accounts by viewModel.accounts.collectAsStateWithLifecycle()
    val isLoading by viewModel.isAccountsLoading.collectAsStateWithLifecycle()
    val error by viewModel.accountsError.collectAsStateWithLifecycle()
    
    when {
        isLoading -> {
            Box(
                modifier = Modifier.fillMaxSize(),
                contentAlignment = Alignment.Center
            ) {
                CircularProgressIndicator(color = Gold)
            }
        }
        error != null -> {
            val errorMsg = error
            AccountsErrorState(error = errorMsg ?: "Unknown error") {
                viewModel.loadAccounts()
            }
        }
        accounts.isEmpty() -> {
            AccountsEmptyState()
        }
        else -> {
            AccountsContent(accounts = accounts)
        }
    }
}

@Composable
fun AccountsContent(accounts: List<com.example.apk.console.models.Account>) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(20.dp)
    ) {
        Text(
            text = "Your Accounts",
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            color = TextPrimary,
            modifier = Modifier.fillMaxWidth()
        )
        
        accounts.forEach { account ->
            AccountCard(account = account)
        }
    }
}

@Composable
fun AccountCard(account: com.example.apk.console.models.Account) {
    GlassCard(
        padding = PaddingValues(20.dp),
        cornerRadius = 16.dp
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Account Icon
            Box(
                modifier = Modifier
                    .size(56.dp)
                    .background(
                        getAccountTypeColor(account.accountType).copy(alpha = 0.2f),
                        androidx.compose.foundation.shape.CircleShape
                    ),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = getAccountTypeIcon(account.accountType),
                    contentDescription = null,
                    tint = getAccountTypeColor(account.accountType),
                    modifier = Modifier.size(24.dp)
                )
            }
            
            // Account Info
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(6.dp)
            ) {
                Text(
                    text = account.bankName,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = TextPrimary
                )
                
                Row(
                    horizontalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    Text(
                        text = account.displayName,
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Medium,
                        color = TextSecondary
                    )
                    
                    account.accountNumber?.let {
                        Text(
                            text = "• $it",
                            fontSize = 14.sp,
                            color = TextSecondary.copy(alpha = 0.5f)
                        )
                    }
                }
            }
            
            // Balance
            Text(
                text = formatCurrency(account.balance),
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = TextPrimary
            )
        }
    }
}

@Composable
private fun AccountsErrorState(error: String, onRetry: () -> Unit) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(60.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Icon(
            imageVector = Icons.Default.Error,
            contentDescription = null,
            tint = Color(0xFFE53935).copy(alpha = 0.7f),
            modifier = Modifier.size(48.dp)
        )
        
        Text(
            text = "Unable to Load Accounts",
            fontSize = 18.sp,
            fontWeight = FontWeight.Bold,
            color = TextPrimary
        )
        
        Text(
            text = error,
            fontSize = 14.sp,
            color = TextSecondary,
            modifier = Modifier.padding(horizontal = 32.dp)
        )
        
        Button(
            onClick = onRetry,
            colors = ButtonDefaults.buttonColors(containerColor = Gold)
        ) {
            Text("Retry", color = com.example.apk.ui.theme.Charcoal)
        }
    }
}

@Composable
private fun AccountsEmptyState() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(60.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Icon(
            imageVector = Icons.Default.CreditCard,
            contentDescription = null,
            tint = TextSecondary.copy(alpha = 0.5f),
            modifier = Modifier.size(48.dp)
        )
        
        Text(
            text = "No Accounts",
            fontSize = 18.sp,
            fontWeight = FontWeight.SemiBold,
            color = TextPrimary
        )
        
        Text(
            text = "Link your bank accounts to see balances",
            fontSize = 14.sp,
            color = TextSecondary,
            modifier = Modifier.padding(horizontal = 32.dp)
        )
    }
}


fun getAccountTypeIcon(type: com.example.apk.console.models.AccountType): androidx.compose.ui.graphics.vector.ImageVector {
    return when (type) {
        com.example.apk.console.models.AccountType.CHECKING -> Icons.Default.CreditCard
        com.example.apk.console.models.AccountType.SAVINGS -> Icons.Default.AccountBalance
        com.example.apk.console.models.AccountType.INVESTMENT -> Icons.Default.TrendingUp
        com.example.apk.console.models.AccountType.CREDIT -> Icons.Default.CreditCard
    }
}

fun getAccountTypeColor(type: com.example.apk.console.models.AccountType): Color {
    return when (type) {
        com.example.apk.console.models.AccountType.CHECKING -> Color(0xFF2196F3)
        com.example.apk.console.models.AccountType.SAVINGS -> Color(0xFF4CAF50)
        com.example.apk.console.models.AccountType.INVESTMENT -> Color(0xFF9C27B0)
        com.example.apk.console.models.AccountType.CREDIT -> Color(0xFFE53935)
    }
}

File: ios_monytix/ios_monytix/console/AccountsTabView.swift
//
//  AccountsTabView.swift
//  ios_monytix
//
//  Created by santosh on 05/01/26.
//

import SwiftUI

struct AccountsTabView: View {
    @ObservedObject var viewModel: MolyConsoleViewModel
    
    private let goldColor = Color(red: 0.831, green: 0.686, blue: 0.216)
    
    var body: some View {
        ScrollView {
            VStack(spacing: 20) {
                if viewModel.isAccountsLoading {
                    ProgressView()
                        .tint(goldColor)
                        .padding(.top, 40)
                } else if let error = viewModel.accountsError {
                    errorState(error)
                } else if viewModel.accounts.isEmpty {
                    emptyState
                } else {
                    accountsContent
                }
            }
            .padding(20)
        }
        .refreshable {
            await viewModel.loadAccounts()
        }
    }
    
    private var accountsContent: some View {
        VStack(alignment: .leading, spacing: 20) {
            Text("Your Accounts")
                .font(.system(size: 20, weight: .bold))
                .foregroundColor(.white)
                .frame(maxWidth: .infinity, alignment: .leading)
            
            ForEach(viewModel.accounts) { account in
                AccountCard(account: account)
            }
        }
    }
    
    private func errorState(_ error: String) -> some View {
        VStack(spacing: 16) {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 48))
                .foregroundColor(.red.opacity(0.7))
            
            Text("Unable to Load Accounts")
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.white)
            
            Text(error)
                .font(.system(size: 14))
                .foregroundColor(.gray.opacity(0.7))
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
            
            Button("Retry") {
                Task {
                    await viewModel.loadAccounts()
                }
            }
            .buttonStyle(GoldButtonStyle())
            .padding(.top, 8)
        }
        .padding(.top, 60)
    }
    
    private var emptyState: some View {
        VStack(spacing: 16) {
            Image(systemName: "creditcard.fill")
                .font(.system(size: 48))
                .foregroundColor(.gray.opacity(0.5))
            
            Text("No Accounts")
                .font(.system(size: 18, weight: .semibold))
                .foregroundColor(.white)
            
            Text("Link your bank accounts to see balances")
                .font(.system(size: 14))
                .foregroundColor(.gray.opacity(0.7))
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
        }
        .padding(.top, 60)
    }
}

// MARK: - Account Card

struct AccountCard: View {
    let account: Account
    
    private let goldColor = Color(red: 0.831, green: 0.686, blue: 0.216)
    
    var body: some View {
        GlassCard(padding: 20, cornerRadius: 16) {
            HStack(spacing: 16) {
                // Account Icon
                ZStack {
                    Circle()
                        .fill(colorForAccountType(account.accountType).opacity(0.2))
                        .frame(width: 56, height: 56)
                    
                    Image(systemName: account.accountType.icon)
                        .font(.system(size: 24, weight: .semibold))
                        .foregroundColor(colorForAccountType(account.accountType))
                }
                
                // Account Info
                VStack(alignment: .leading, spacing: 6) {
                    Text(account.bankName)
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(.white)
                    
                    HStack(spacing: 8) {
                        Text(account.accountType.displayName)
                            .font(.system(size: 14, weight: .medium))
                            .foregroundColor(.gray.opacity(0.7))
                        
                        if let accountNumber = account.accountNumber {
                            Text("• \(accountNumber)")
                                .font(.system(size: 14))
                                .foregroundColor(.gray.opacity(0.5))
                        }
                    }
                }
                
                Spacer()
                
                // Balance
                VStack(alignment: .trailing, spacing: 4) {
                    Text(formatCurrency(account.balance))
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(.white)
                }
            }
        }
    }
    
    private func colorForAccountType(_ type: AccountType) -> Color {
        switch type {
        case .checking:
            return .blue
        case .savings:
            return .green
        case .investment:
            return .purple
        case .credit:
            return .red
        }
    }
    
    private func formatCurrency(_ amount: Double) -> String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .currency
        formatter.currencySymbol = "₹"
        formatter.maximumFractionDigits = 0
        return formatter.string(from: NSNumber(value: amount)) ?? "₹0"
    }
}

#Preview {
    let authService = AuthService()
    return AccountsTabView(viewModel: MolyConsoleViewModel(authService: authService))
        .background(Color(red: 0.18, green: 0.18, blue: 0.18))
}

